### PR TITLE
feat: Remove Slacks auto prefixing of url protocols

### PR DESCRIPTION
**Pull request by trygveaa**
When you write multiple words separated by dots, or email adresses, Slack will turn those into links. Wee-slack will display this as `text (url)`, where url is just the text with http:// or mailto: prefixed. This makes the text hard to read, and might not even make sense (e.g. when writing server names or user@hostname).

When you receive such an url, the alt text of the url will be the original text. This means that we can check if the url is just the alt text with a protocol prefixed. If it is, we should just print the alt text and ignore the url.

I guess some people may want the urls, so they are able to click the links. And since it removes information coming from slack, I made it into an option instead of making it the default.

### CODE REVIEW

File: wee_slack.py [hunk 1]
```diff
@@ -2587,7 +2587,11 @@ def unfurl_ref(ref, ignore_alt_text=False):
                 display_text = ref.split('|')[1]
             else:
                 url, desc = ref.split('|', 1)
-                display_text = "{} ({})".format(url, desc)
+                if (config.unfurl_prevent_auto_linking and
+                        re.match(r"^\w+:(//)?{}$".format(desc), url)):
```
Comment (tfheen) [inline wee_slack.py]: Maybe use `re.escape(desc)` to ensure regex metacharacters aren't treated specially?

**Comment by auscompgeek** [comment]
Could we also allow for an option to hide the other way (i.e. hide the user-typed URL if Slack auto-prefixed the link with the relevant protocol)?

File: wee_slack.py (hunk 1 above)
Comment (trygveaa) [reply wee_slack.py]: Definitely, should have thought of that.

**Comment by trygveaa** [comment]
> Could we also allow for an option to hide the other way (i.e. hide the user-typed URL if Slack auto-prefixed the link with the relevant protocol)?

Sure, I've added support for that now, and changed the option (and the help text) so you can choose between the three alternatives.

**Pull request merged**